PR: https://github.com/odoo/odoo/pull/173273

From: 88569c97efbb534679700e8c7d56231aa1934b4d
From: Jitendra Prajapat
Date: 2024-08-09 17:05:55

Breaking data model changes scores: del:7 + add:0, change matches:
-    _inherit = "pos.payment"
-    paytm_authcode = fields.Char('Paytm APPR Code')
-    paytm_issuer_card_no = fields.Char('Paytm Issue Mask Card No.')
-    paytm_issuer_bank = fields.Char('Paytm Issuer Bank')
-    paytm_payment_method = fields.Char('Paytm Payment Method')
-    paytm_reference_no = fields.Char('Paytm Merchant Reference No.')
-    paytm_card_scheme = fields.Char('Paytm Card Scheme')

Total Changes: 16

[IMP] pos*: made pos_payment fields common

pos*: point_of_sale, pos_razorpay, pos_paytm

Before this commit:
===================
There were separate fields for pos_payment for each module, such as pos_paytm,
pos_razorpay, and pos_mercury. Each field stored the same data for each module
individually.

After this commit:
================
A common field for pos_payment are created to store data based on the module,
simplifying the data management and making it easier to display on the UI side.

task- 4011453

Related Upgrade PR: https://github.com/odoo/upgrade/pull/6272

closes odoo/odoo#173273

Related: odoo/upgrade#6272
Signed-off-by: Adrien Guilliams (adgu) <adgu@odoo.com>

================================= pseudo patch: =================================

--- a/addons/pos_paytm/models/__init__.py
+++ b/addons/pos_paytm/models/__init__.py
@@ -1,2 +1 @@
 from . import pos_payment_method
-from . import pos_payment

--- a/addons/pos_paytm/models/pos_payment.py
+++ b/None
@@ -1,15 +0,0 @@
-# -*- coding: utf-8 -*-
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import api, models, fields
-
-class PosPayment(models.Model):
-
-    _inherit = "pos.payment"
-
-    paytm_authcode = fields.Char('Paytm APPR Code')
-    paytm_issuer_card_no = fields.Char('Paytm Issue Mask Card No.')
-    paytm_issuer_bank = fields.Char('Paytm Issuer Bank')
-    paytm_payment_method = fields.Char('Paytm Payment Method')
-    paytm_reference_no = fields.Char('Paytm Merchant Reference No.')
-    paytm_card_scheme = fields.Char('Paytm Card Scheme')
